### ISSUE TITLE
Expose WatchEvents on demand

### ISSUE THREAD

**Issue by clux** [clux/kube-rs#5]
A couple of options, not sure what's the nicest interface yet:

- On demand events since last "reconcile"
- Events as they occur through callbacks

First is easy to do, just queue up, then pop and return on demand.
Latter would probably block the polling thread/owner (since that's the thing that runs the watcher).

**Comment by clux** [comment]
We now cache the events in the Reflector as well. So it should be possible to build a reconcile fn. Ideally, call it from the `thread` that owns the Reflector. Same loop is an easy start.

```toml
kube = { git = "https://github.com/clux/kube-rs", rev = "6cd8bfe4f8159036055c7d0cabcdf0d41c300b04" }
```

See https://github.com/clux/kube-rs/blob/6cd8bfe4f8159036055c7d0cabcdf0d41c300b04/examples/pod_reconcile.rs#L36-L64 for example usage.

**Comment by clux** [comment]
Discussion still to be had about the best way to expose it #6 - but initial feature is available in `0.4.0`.
